huaweicloud/services/rds/resource_huaweicloud_rds_instance.go

```diff
@@ -1035,19 +1050,25 @@ func updateRdsInstanceVolumeSize(d *schema.ResourceData, client *golangsdk.Servi
 	return nil
 }
 
-func updateRdsInstanceBackpStrategy(d *schema.ResourceData, client *golangsdk.ServiceClient, instanceID string) error {
+func updateRdsInstanceBackupStrategy(d *schema.ResourceData, client *golangsdk.ServiceClient, instanceID string) error {
 	if !d.HasChange("backup_strategy") {
 		return nil
 	}
-
+	backupStrategy, errs := backups.Get(client, instanceID).Extract()
+	if errs != nil {
+		return fmt.Errorf("error getting RDS backup strategy: %s", errs)
+	}
 	backupRaw := d.Get("backup_strategy").([]interface{})
 	rawMap := backupRaw[0].(map[string]interface{})
 	keepDays := rawMap["keep_days"].(int)
-
+	period := rawMap["period"].(string)
+	if period == "" {
+		period = backupStrategy.Period
```

Comment:
can we set period to "1,2,3,4,5,6,7" when the field is empty?

Reply:
ok, i have modified